evm_loader/program/src/payment.rs

```diff
@@ -0,0 +1,70 @@
+//! `evm_loader` program payment module.
+
+use solana_program::{
+    account_info::AccountInfo,
+    entrypoint::ProgramResult,
+    program::invoke,
+    program_error::ProgramError,
+    pubkey::Pubkey,
+    system_instruction,
+};
+
+// use std::str::FromStr;
+
+// TODO set collateral pool base address
+// const COLLATERAL_POOL_BASE: &str = "4sW3SZDJB7qXUyCYKA7pFL8eCTfm3REr8oSiKkww7MaT";
+const COLLATERAL_SEED_PREFIX: &str = "collateral_seed_";
+const PAYMENT_TO_COLLATERAL_POOL: u64 = 1000;
+
+/// Checks collateral accounts for the Ethereum transaction execution.
+#[allow(clippy::unnecessary_wraps)]
+#[allow(unused_variables)]
+pub fn check_collateral_account(
+    program_id: &Pubkey,
+    // WARNING Only for tests when base is random
+    collateral_pool_base: &AccountInfo,
+    collateral_pool_sol_info: &AccountInfo,
+    collateral_pool_index: usize
+) -> ProgramResult {
+    debug_print!("program_id {}", program_id);
+    debug_print!("collateral_pool_sol_info {:?}", collateral_pool_sol_info);
+    debug_print!("collateral_pool_index {}", collateral_pool_index);
+
+    // Owner of collateral_pool_sol_info is system: 11111111111111111111111111111111
+    /*if collateral_pool_sol_info.owner != program_id {
+        debug_print!("Wrong collateral pool owner {}", *collateral_pool_sol_info.owner);
+        debug_print!("Must be program_id {}", program_id);
+        return Err(ProgramError::InvalidArgument);
+    }*/
```

Comment:
Why this is commented? Delete it if you don't need it.

Reply:
These lines would be returned when a predefined collateral owner account would be used.